### PR TITLE
[labs/nextjs] Bump Peer Dependencies for Next 15 Support

### LINKED ISSUE #4812: [labs/nextjs] Add support for Next.js 15
### Should this be an RFC?

- [x] This is not a substantial change

### Which package is this a feature request for?

Other/unknown (please mention in description)

### Description

Next.js 15 was just released (https://nextjs.org/blog/next-15). I was curious enough to try lit on Next.js 15, but it gave me dependency resolution error on @lit-labs/nextjs. 
```
npm error code ERESOLVE
npm error ERESOLVE unable to resolve dependency tree
npm error
npm error While resolving: test@0.1.0
npm error Found: next@15.0.1
npm error node_modules/next
npm error   next@"15.0.1" from the root project
npm error
npm error Could not resolve dependency:
npm error peer next@"13 || 14" from @lit-labs/nextjs@0.2.0
npm error node_modules/@lit-labs/nextjs
npm error   @lit-labs/nextjs@"*" from the root project
npm error
npm error Fix the upstream dependency conflict, or retry
npm error this command with --force or --legacy-peer-deps
npm error to accept an incorrect (and potentially broken) dependency resolution.
```
I was able to get it running for a very basic use case via bypassing dependency checking. Are there anything released from Next.js 15 could potentially break the @lit-labs/nextjs packages? 

Below is how I used the @lit-labs/nextjs package in a Next.js 15 app (app router).
**Steps:**
1. Install app router Nextjs like normal. Add `--legacy-peer-deps` to bypass warnings when installing @lit-labs/nextjs.
```
npx create-next-app@latest test
cd test
npm i lit @lit/react
npm i @lit-labs/nextjs --legacy-peer-deps
```

2. Configure next.config.ts
```
import type { NextConfig } from "next";
import withLitSSR from "@lit-labs/nextjs";

const nextConfig: NextConfig = {
  /* config options here */
  reactStrictMode: true,
};

export default withLitSSR({
  addDeclarativeShadowDomPolyfill: true,
})(nextConfig);
```

3. Add web component in page.tsx
```
"use client";

import { LitElement, html } from "lit";
import { customElement } from "lit/decorators.js";
import { createComponent } from "@lit/react";
import React from "react";

@customElement("my-element")
export class MyElement extends LitElement {
  render() {
    return html` <div>Hello from my Lit Element!</div> `;
  }
}

const HelloComponent = createComponent({
  tagName: "my-element",
  elementClass: MyElement,
  react: React,
});

export default function Home() {
  return (
    <HelloComponent/>
  );
}
```

4. Run `npm run dev` and web component is displayed.

**Potential issues:**
- Turbopack seems not working after configuring lit SSR in step 2.
- No other notable issues found for now

### Alternatives and Workarounds

Use `--legacy-peer-deps` to bypass warning during installation.

### PR DESCRIPTION
**Pull request by JamesIves**
Fixes #4812 

## Description

Bumps peer dependencies so labs/nextjs works with Next.js 15. Also adds two sample apps for Next.js 15 (one for app router, another for pages) by copying and pasting the v14 samples and running the Next.js updater tool on them. 

I took a pass at validating the SSR functionality and everything seemed fine, but would appreciate a sanity check. 

Happy to make any changes based on feedback!

### CODE REVIEW

**Review by justinfagnani — COMMENTED** [review]
Thanks @JamesIves!

LGTM - wdyt @augustjk ?

**Review by justinfagnani — APPROVED** [review]

File: examples/nextjs-v15-app/next.config.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+const withLitSSR = require('@lit-labs/nextjs')({
+  addDeclarativeShadowDomPolyfill: true,
+});
+
+/** @type {import('next').NextConfig} */
+const nextConfig = {
+  reactStrictMode: true,
+  swcMinify: true,
```
Comment (augustjk) [inline examples/nextjs-v15-app/next.config.js:8]: I'm seeing the following when running so they must've removed the option in this version.
```
 ⚠ Invalid next.config.js options detected: 
 ⚠     Unrecognized key(s) in object: 'swcMinify'
 ⚠ See more info here: https://nextjs.org/docs/messages/invalid-next-config
```

```suggestion
```

File: examples/nextjs-v15/next.config.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+const withLitSSR = require('@lit-labs/nextjs')({
+  addDeclarativeShadowDomPolyfill: true,
+});
+
+/** @type {import('next').NextConfig} */
+const nextConfig = {
+  reactStrictMode: true,
+  swcMinify: true,
```
Comment (augustjk) [inline examples/nextjs-v15/next.config.js:8]: Same note as above.
```suggestion
```

**Review by augustjk — APPROVED** [review]
Thanks! This examples seem to work fine with SSR in the new version. Just a couple suggestions to remove the now-invalid next config option.

File: examples/nextjs-v15/next.config.js (hunk 2 above)
Comment (JamesIves) [reply examples/nextjs-v15/next.config.js]: Done!

**Comment by JamesIves** [comment]
@augustjk @justinfagnani Should be all good to go now

**Pull request merged**